vcx/libvcx/src/messages/agent_utils.rs

```diff
@@ -514,6 +522,33 @@ pub fn send_message_to_agency(message: &A2AMessage, did: &str) -> VcxResult<Vec<
     parse_response_from_agency(&response, &settings::get_protocol_type())
 }
 
+pub fn create_pub_agent() -> VcxResult<()> {
```

Comment:
I think this should return `AgentInfo` (well, now it's `CloudAgentInfo`) and there should be also way for to "load" or "deserialize" this CloudAgentInfo of public agent - if I restart running process, I need to be able to able to to call `save_to_cache` without creating new public agent (in context of the code below)